tests/integration/test_clock_drift.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta, datetime
+
+from sentry_sdk.envelope import Envelope
+
+
+def mock_transaction(timestamp):
+    return {
+        "type": "transaction",
+        "timestamp": timestamp.isoformat(),
+        "start_timestamp": (timestamp - timedelta(seconds=2)).isoformat(),
+        "spans": [],
+        "contexts": {
+            "trace": {
+                "op": "http",
+                "trace_id": "a0fa8803753e40fd8124b21eeb2986b5",
+                "span_id": "968cff94913ebb07",
+            }
+        },
+        "transaction": "/hello",
+    }
+
+
+def test_clock_drift_applied_when_timestamp_is_too_old(mini_sentry, relay):
+    relay = relay(mini_sentry)
+    project_id = 42
+    mini_sentry.add_basic_project_config(project_id)
+
+    now = datetime.utcnow()
+    one_month_ago = now - timedelta(days=30)
+
+    envelope = Envelope(
+        headers={"sent_at": one_month_ago.strftime("%Y-%m-%dT%H:%M:%S.%fZ")}
+    )
+    envelope.add_transaction(mock_transaction(one_month_ago))
+
+    relay.send_envelope(project_id, envelope, headers={"Accept-Encoding": "gzip"})
+
+    transaction_event = mini_sentry.captured_events.get(
+        timeout=1
+    ).get_transaction_event()
+    error_name, error_metadata = transaction_event["_meta"]["timestamp"][""]["err"][0]
+    assert error_name == "clock_drift"
```

Comment:
Should we also assert on the actual timestamps in the event? (Same for tests below).

Reply:
I was considering it, my only problem was that `received_at` seems to be inferred by Relay on receive time, so I might have to either mock things or freeze time but it seems like this is not possible in an integration test.

Reply:
What I know is that `sdk_time` will be `one_month_ago` and `server_time` will be `received_at` and the new timestamp is + the difference of the above.

Reply:
I opted for a middle ground, I do test for `>` in the timestamps which should be normalized, and `=` when the timestamps must remain the same.

Reply:
If there was a way to freeze or hardcode the `received_at` time, I could compute the exact test expectations. 

If we can't, to make the test more precise, we could assert `transaction_event["timestamp"] == now` within some bounds but I am not super confident to define them since they are dependent on the time it takes for the integration test to run and I would not want to introduce a flaky test.

Reply:
I wouldn't exactly assert, but if you compare to the `now` value taken at the beginning of the test, that should be good, additionally you could also take another "now" value at the end and check whether the expected value is between those. I wouldn't try to freeze time etc. that will just break at some point. Being within a few seconds should be good enough.